Stream capture error in ROCm tests

https://jenkins.preferred.jp/job/chainer/job/cupy_master/TEST=rocm-4-0,label=mnj-mi50/1919/console

```
 E   RuntimeError: streamIsCapturing is supported since CUDA 10.1+
```

--- a/cupy/cuda/memory.pyx
+++ b/cupy/cuda/memory.pyx
@@ -407,7 +407,10 @@ cdef class MemoryPointer:
 
         """
         stream_ptr = stream_module.get_current_stream_ptr()
-        if runtime.streamIsCapturing(stream_ptr):
+        if (
+            not runtime._is_hip_environment
+            and runtime.streamIsCapturing(stream_ptr)
+        ):
             raise RuntimeError(
                 'the current stream is capturing, so synchronous API calls '
                 'are disallowed')
@@ -451,7 +454,10 @@ cdef class MemoryPointer:
 
         """
         stream_ptr = stream_module.get_current_stream_ptr()
-        if runtime.streamIsCapturing(stream_ptr):
+        if (
+            not runtime._is_hip_environment
+            and runtime.streamIsCapturing(stream_ptr)
+        ):
             raise RuntimeError(
                 'the current stream is capturing, so synchronous API calls '
                 'are disallowed')
@@ -475,7 +481,10 @@ cdef class MemoryPointer:
             stream_ptr = stream_module.get_current_stream_ptr()
         else:
             stream_ptr = stream.ptr
-        if runtime.streamIsCapturing(stream_ptr):
+        if (
+            not runtime._is_hip_environment
+            and runtime.streamIsCapturing(stream_ptr)
+        ):
             raise RuntimeError(
                 'the current stream is capturing, so H2D transfers '
                 'are disallowed')
@@ -543,7 +552,10 @@ cdef class MemoryPointer:
 
         """
         stream_ptr = stream_module.get_current_stream_ptr()
-        if runtime.streamIsCapturing(stream_ptr):
+        if (
+            not runtime._is_hip_environment
+            and runtime.streamIsCapturing(stream_ptr)
+        ):
             raise RuntimeError(
                 'the current stream is capturing, so synchronous API calls '
                 'are disallowed')
@@ -567,7 +579,10 @@ cdef class MemoryPointer:
             stream_ptr = stream_module.get_current_stream_ptr()
         else:
             stream_ptr = stream.ptr
-        if runtime.streamIsCapturing(stream_ptr):
+        if (
+            not runtime._is_hip_environment
+            and runtime.streamIsCapturing(stream_ptr)
+        ):
             raise RuntimeError(
                 'the current stream is capturing, so D2H transfers '
                 'are disallowed')
@@ -591,7 +606,10 @@ cdef class MemoryPointer:
 
         """
         stream_ptr = stream_module.get_current_stream_ptr()
-        if runtime.streamIsCapturing(stream_ptr):
+        if (
+            not runtime._is_hip_environment
+            and runtime.streamIsCapturing(stream_ptr)
+        ):
             raise RuntimeError(
                 'the current stream is capturing, so synchronous API calls '
                 'are disallowed')

--- a/cupy_backends/cuda/api/runtime.pyx
+++ b/cupy_backends/cuda/api/runtime.pyx
@@ -864,6 +864,8 @@ cpdef streamWaitEvent(intptr_t stream, intptr_t event, unsigned int flags=0):
 
 
 cpdef streamBeginCapture(intptr_t stream, int mode=streamCaptureModeRelaxed):
+    if _is_hip_environment:
+        raise RuntimeError('streamBeginCapture is not supported in ROCm')
     # TODO(leofang): check and raise if stream == 0?
     if CUPY_CUDA_VERSION < 10010:
         raise RuntimeError('streamBeginCapture is supported since CUDA 10.1+')
@@ -876,6 +878,8 @@ cpdef streamBeginCapture(intptr_t stream, int mode=streamCaptureModeRelaxed):
 cpdef intptr_t streamEndCapture(intptr_t stream) except? 0:
     # TODO(leofang): check and raise if stream == 0?
     cdef Graph g
+    if _is_hip_environment:
+        raise RuntimeError('streamEndCapture is not supported in ROCm')
     if CUPY_CUDA_VERSION < 10010:
         raise RuntimeError('streamEndCapture is supported since CUDA 10.1+')
     with nogil:
@@ -886,6 +890,8 @@ cpdef intptr_t streamEndCapture(intptr_t stream) except? 0:
 
 cpdef bint streamIsCapturing(intptr_t stream) except*:
     cdef StreamCaptureStatus s
+    if _is_hip_environment:
+        raise RuntimeError('streamIsCapturing is not supported in ROCm')
     if CUPY_CUDA_VERSION < 10010:
         raise RuntimeError('streamIsCapturing is supported since CUDA 10.1+')
     with nogil:

--- a/cupy_backends/cuda/libs/cublas.pyx
+++ b/cupy_backends/cuda/libs/cublas.pyx
@@ -470,7 +470,7 @@ cpdef setStream(intptr_t handle, size_t stream):
     # https://docs.nvidia.com/cuda/cublas/index.html#CUDA-graphs
     # Before we come up with a robust strategy to test the support conditions,
     # we disable this functionality.
-    if runtime.streamIsCapturing(stream):
+    if not runtime._is_hip_environment and runtime.streamIsCapturing(stream):
         raise NotImplementedError(
             'calling cuBLAS API during stream capture is currently '
             'unsupported')

--- a/cupy_backends/cuda/libs/cudnn.pyx
+++ b/cupy_backends/cuda/libs/cudnn.pyx
@@ -836,7 +836,7 @@ cpdef destroy(intptr_t handle):
 cpdef setStream(intptr_t handle, size_t stream):
     # TODO(leofang): The support of stream capture is not mentioned at all in
     # the cuDNN docs (as of CUDA 11.5), so we disable this functionality.
-    if runtime.streamIsCapturing(stream):
+    if not runtime._is_hip_environment and runtime.streamIsCapturing(stream):
         raise NotImplementedError(
             'calling cuDNN API during stream capture is currently '
             'unsupported')

--- a/cupy_backends/cuda/libs/curand.pyx
+++ b/cupy_backends/cuda/libs/curand.pyx
@@ -115,7 +115,7 @@ cpdef int getVersion() except? -1:
 cpdef setStream(size_t generator, size_t stream):
     # TODO(leofang): The support of stream capture is not mentioned at all in
     # the cuRAND docs (as of CUDA 11.5), so we disable this functionality.
-    if runtime.streamIsCapturing(stream):
+    if not runtime._is_hip_environment and runtime.streamIsCapturing(stream):
         raise NotImplementedError(
             'calling cuRAND API during stream capture is currently '
             'unsupported')

--- a/cupy_backends/cuda/libs/cusolver.pyx
+++ b/cupy_backends/cuda/libs/cusolver.pyx
@@ -1136,7 +1136,7 @@ cpdef spDestroy(intptr_t handle):
 cpdef setStream(intptr_t handle, size_t stream):
     # TODO(leofang): The support of stream capture is not mentioned at all in
     # the cuSOLVER docs (as of CUDA 11.5), so we disable this functionality.
-    if runtime.streamIsCapturing(stream):
+    if not runtime._is_hip_environment and runtime.streamIsCapturing(stream):
         raise NotImplementedError(
             'calling cuSOLVER API during stream capture is currently '
             'unsupported')

--- a/cupy_backends/cuda/libs/cusparse.pyx
+++ b/cupy_backends/cuda/libs/cusparse.pyx
@@ -1596,7 +1596,7 @@ cpdef setStream(intptr_t handle, size_t stream):
     # https://docs.nvidia.com/cuda/cusparse/index.html#optimization-notes
     # Before we come up with a robust strategy to test the support conditions,
     # we disable this functionality.
-    if runtime.streamIsCapturing(stream):
+    if not runtime._is_hip_environment and runtime.streamIsCapturing(stream):
         raise NotImplementedError(
             'calling cuSPARSE API during stream capture is currently '
             'unsupported')

